### PR TITLE
[feat] engine: add hex.pm package search

### PR DESCRIPTION
**Pull request by dkuku**
## What does this PR do?
Add search for elixir/erlang/gleam packages on hex.pm
Code is copied from the npm search. I'm just not sure how to test it.
There is no documentation page but the code is open source so I linked to the search controller as api link

image

### CODE REVIEW

**Comment by return42** [comment]
Thanks for your PR / there is an issue [reported by the CI](https://github.com/searxng/searxng/actions/runs/8941676083/job/24563007683?pr=3455) .. you can fix it by `make format.python` .. see our [Development Quickstart guide](https://docs.searxng.org/dev/quickstart.html#development-quickstart) .. or if it is OK for you, I can fix the issue when I review your PR.

**Comment by dkuku** [comment]
if you can format it then it would be great. I just left home.

File: searx/engines/hex.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+def request(query: str, params):
+    args = urlencode({"page": params["pageno"], "search": query})
+    params["url"] = search_api + args
+    return params
+
+
+def response(resp):
+    results = []
+    content = resp.json()
+    for package in content:
```
Comment (Bnyro) [inline searx/engines/hex.py]:
```suggestion
    for package in resp.json():
```
nit: this can be inlined

File: searx/engines/hex.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    "require_api_key": False,
+    "results": "JSON",
+}
+
+categories = ["it", "packages"]
+
+
+# engine dependent config
+paging = True
+search_api = "https://hex.pm/api/packages/?"
```
Comment (Bnyro) [inline searx/engines/hex.py]:
```suggestion
search_url = "https://hex.pm/api/packages/"
```
the question mark should only be added in the `request` function for readability

File: searx/engines/hex.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+
+def request(query: str, params):
+    args = urlencode({"page": params["pageno"], "search": query})
+    params["url"] = search_api + args
```
Comment (Bnyro) [inline searx/engines/hex.py]:
```suggestion
    params["url"] = f"{search_url}?{args}"
```

Comment (dkuku) [reply searx/engines/hex.py]: Looks much better, thanks

File: searx/engines/hex.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+"""hex.pm"""
+
+from urllib.parse import urlencode
+from dateutil import parser
+
+
+about = {
+    "website": "https://hex.pm/",
+    "wikidata_id": None,
+    "official_api_documentation": "https://github.com/hexpm/hexpm/blob/main/lib/hexpm_web/controllers/api/package_controller.ex",
```
Comment (Bnyro) [inline searx/engines/hex.py:12]:
```suggestion
    "official_api_documentation": "https://github.com/hexpm/hexpm/blob/main/lib/hexpm_web/controllers/api/package_controller.ex", # pylint: disable=line-too-long
```
fixes the CI / linter

**Review by return42 — APPROVED** [review]
> if you can format it then it would be great. I just left home.

I removed the timeout [diff](https://github.com/searxng/searxng/compare/3b889b7f4c7dfc431fc722bb7b00b938b772756c..d9b63916eef408231355448a8aeae626f94ee7a1), squashed the commits and modified the [commit message](https://github.com/searxng/searxng/pull/3455/commits/d9b63916eef408231355448a8aeae626f94ee7a1)

**Review by Bnyro — APPROVED** [review]
LGTM, thanks @dkuku :+1:

**Comment by dkuku** [comment]
Thanks for the review

**Pull request merged**